### DISCUSSION TITLE
var(--spacing-2) removed, now using calc... using in style tag in .svelte file

### DISCUSSION THREAD

**Discussion by Dudek-AMS** [tailwindlabs/tailwindcss#15027] [Help] [answered]
How am I supposed to use --spacing-* or similiar vars in a .svelte file? Do I really have to calc it all the on my own?
`@apply p-2` i.e. only works, when i explicetly import "tailwindcss" inside my component

**Comment by Dudek-AMS** [comment]
And importing tailwindcss is bloating the components css like hell

**Comment by philipp-spiess** [comment, marked as answer]
@Dudek-AMS Hey! We have a feature for this to import the default theme without bloating the CSS file, you can try this:

```css
@import "tailwindcss/theme" theme(reference);

.foo {
  @apply p-2;
}
```

This will pull in the theme but won't emit any CSS variables when doing so and was built specifically for this use case. You can substitute the import with your own Tailwind theme configuration CSS file as well.

**Reply by Dudek-AMS** [reply]
is there a way to autoinject this?

**Comment by Dudek-AMS** [comment]
For all people interested I was able to autoinject the `@import` in svelte.config.js

```js

import adapter from '@sveltejs/adapter-node';
import { vitePreprocess } from '@sveltejs/vite-plugin-svelte';

/** @type {import('@sveltejs/kit').Config} */
const config = {

	preprocess: [
		{
			name: 'injectTailwind',
			style({ content, filename }) {
				console.log('style', filename);
				if(!content.includes('@import "tailwindcss/theme"')) {
					content = '@import "tailwindcss/theme" theme(reference);\n' + content;
				}
				return { code: content, map: null };
			}
		},
		vitePreprocess()
	],
	kit: {
		adapter: adapter(),
		csrf: {
			checkOrigin: false
		}
	}
};

export default config;
```

**Comment by oneezy** [comment]
@philipp-spiess is there a cleaner way to do this possibly? asking because this is could be a ton of boilerplate.

I created an [issue](https://github.com/sveltejs/svelte/issues/14408) in **sveltejs/svelte** but if it could be fixed in **tw4** and _just work_ it would be a lot better developer experience
